frontend/src/bundles/auth/store/actions.ts

```diff
@@ -1,33 +1,53 @@
-import { createAsyncThunk } from '@reduxjs/toolkit';
+import { createAsyncThunk, unwrapResult } from '@reduxjs/toolkit';
 
 import { type AsyncThunkConfig } from '~/bundles/common/types/types.js';
 import {
+    type UserGetCurrentResponseDto,
     type UserSignInRequestDto,
     type UserSignInResponseDto,
     type UserSignUpRequestDto,
     type UserSignUpResponseDto,
 } from '~/bundles/users/users.js';
+import { StorageKey } from '~/framework/storage/enums/enums.js';
 
 import { name as sliceName } from './slice.js';
 
 const signIn = createAsyncThunk<
     UserSignInResponseDto,
     UserSignInRequestDto,
     AsyncThunkConfig
->(`${sliceName}/sign-in`, (signInPayload, { extra }) => {
-    const { authApi } = extra;
-
-    return authApi.signIn(signInPayload);
+>(`${sliceName}/sign-in`, async (signInPayload, { extra, dispatch }) => {
+    const { authApi, storage } = extra;
+    const response = await authApi.signIn(signInPayload);
+    if (response.token) {
+        await storage.set(StorageKey.TOKEN, response.token);
+        await dispatch(loadCurrentUser()).then(unwrapResult);
```

Comment:
Why do we need to dispatch loadCurrentUser right after sign-in and sign-up?

Reply:
That's in the task requirements. I did this way for correct mistakes output if loadCurrent fails, sign-in fails as well.

Reply:
When I started working on the task it only returned token, so loading user was required, now I am not so sure, maybe we wait for mentor's opinion?

Reply:
Also, response types for sign-in and sign-up are different, they have similar properties to current user but they may change in the future. Oh, and I noticed that sign-in doesn't have fullFame.

Reply:
You should retrieve the user after sign-in. Get current user is needed for page refresh, for example. There is no need to do the same job twice